units/projectile.gd

```diff
@@ -1,6 +1,22 @@
 extends Area2D
 
 const SPEED = 750
+var projectile_effects: Array[ProjectileEffect] = []
+
+
+func spawn(owner: Node, spawn_with_projectile_effects: Array[ProjectileEffect]):
+	# Execute any spawn modification effects for the projectile
+	projectile_effects = spawn_with_projectile_effects
+	for effect in projectile_effects:
+		effect.modify_creation(owner)
+	# Spawn the default projectile
+	# TODO:: Projectile spawn modification may want to prevent the default from spawning
+	#    how do?
+	owner.add_child(self)
+	projectile_effects = projectile_effects
 
 func _physics_process(delta):
-	position += transform.x * SPEED * delta
+	var position_modifier = transform.x * SPEED * delta
+	for effect in projectile_effects:
```

Comment:
Will all effects get applied every tick? Or will some only apply every x-seconds / ticks? Thinking i.e. a projectile that splits periodically. Or explodes after a certain amount of time.
Or would we have a "timer" effect that chains with something that will happen after that time? 🤔

Reply:
I think whether or not the effect needs to o something should be a responsibility of the effect.

I've since pushed a few changes that basically involve just passing everything relate to the operation down the callstack, which is kinda ugly but it does let us obfuscate the underlying details